LobbyServer2/LobbyServer/LobbyServerProtocol.cs

```diff
@@ -190,6 +193,36 @@ public void RefreshGroup(bool resetReadyState = false)
             Send(update);
         }
 
+        private void HandleGroupPromoteRequest(GroupPromoteRequest message)
+        {
+            LobbyPlayerGroupInfo info = GroupManager.GetGroupInfo(AccountId);
+            info.Members.Find(m => m.MemberDisplayName == message.Name).IsLeader = true;
+            info.Members.Find(m => m.AccountID == AccountId).IsLeader = false;
```

Comment:
I guess we need to validate that the requester is current leader and that the new leader is in the group... at least someday we need.

But Leader property in GroupInfo seems to work incorrectly now, probably it should just iterate over members and find first IsLeader